core/storage/topic/gae_models_test.py

```diff
@@ -0,0 +1,167 @@
+# coding: utf-8
+#
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for Topic model."""
+
+from constants import constants
+from core.domain import topic_domain
+from core.domain import topic_services
+from core.platform import models
+from core.tests import test_utils
+import feconf
+
+(topic_models,) = models.Registry.import_models([models.NAMES.topic])
+
+
+class TopicModelUnitTests(test_utils.GenericTestBase):
+    """Tests the TopicModel class."""
+
+    def test_that_subsidiary_models_are_created_when_new_model_is_saved(self):
+        """Tests the _trusted_commit() method."""
+
+        # Topic is created but not committed/saved.
+        topic = topic_models.TopicModel(
+            id='topic_id',
+            name='topic_name',
+            subtopic_schema_version=feconf.CURRENT_SUBTOPIC_SCHEMA_VERSION,
+            next_subtopic_id=1,
+            language_code='en'
+        )
+        # We check that topic has not been saved before calling commit().
+        self.assertIsNone(topic_models.TopicModel.get_by_name('topic_name'))
+        # We call commit() expecting that _trusted_commit works fine
+        # and saves topic to datastore.
+        topic.commit(
+            committer_id=feconf.SYSTEM_COMMITTER_ID,
+            commit_message='Created new topic',
+            commit_cmds=[{'cmd': topic_domain.CMD_CREATE_NEW}]
+        )
+        # Now we check that topic is not None and that actually
+        # now topic exists, that means that commit() worked fine.
+        self.assertIsNotNone(topic_models.TopicModel.get_by_name('topic_name'))
+
+    def test_get_by_name(self):
+        topic = topic_domain.Topic.create_default_topic(
+            topic_id='topic_id',
+            name='topic_test'
+        )
+        topic_services.save_new_topic(feconf.SYSTEM_COMMITTER_ID, topic)
+        self.assertEqual(
+            topic_models.TopicModel.get_by_name(topic.name).name,
+            topic.name
+        )
+        self.assertEqual(
+            topic_models.TopicModel.get_by_name(topic.name).id,
+            topic.id
```

Comment:
Ditto.

If you like, it's also fine to have class-level string constants for TOPIC_ID and TOPIC_NAME and use these both when constructing the topic (in lines 58-59) and in the test checks. This is still OK because the expected value is a primitive type. In fact it's probably better to do it this way since it's better to use named constants than raw strings.